process/factory/shard/interceptorsContainerFactory.go

```diff
@@ -309,15 +308,32 @@ func (icf *interceptorsContainerFactory) generateUnsignedTxsInterceptors() ([]st
 }
 
 func (icf *interceptorsContainerFactory) createOneUnsignedTxInterceptor(identifier string) (process.Interceptor, error) {
-	uTxStorer := icf.store.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	//TODO replace the nil tx validator with white list validator
+	txValidator, err := dataValidators.NewNilTxValidator()
+	if err != nil {
+		return nil, err
+	}
+
+	argProcessor := &processor.ArgTxInterceptorProcessor{
+		ShardedDataCache: icf.dataPool.UnsignedTransactions(),
+		TxValidator:      txValidator,
+	}
+	txProcessor, err := processor.NewTxInterceptorProcessor(argProcessor)
+	if err != nil {
+		return nil, err
+	}
 
-	interceptor, err := unsigned.NewUnsignedTxInterceptor(
+	txFactory, err := interceptorFactory.NewInterceptedDataFactory(
+		icf.argInterceptorFactory,
+		interceptorFactory.InterceptedUnsignedTx,
+	)
```

Comment:
you need to check the error for creating the interceptor data factory as well

Reply:
Good catch, found another one. Fixed.